lib/membrane_rtc_engine/endpoints/webrtc/rtp_connection_allocator.ex

```diff
@@ -75,14 +92,16 @@ defmodule Membrane.RTC.Engine.Endpoint.WebRTC.RTPConnectionAllocator do
   @impl true
   def handle_cast({:bandwidth_estimation, estimation}, state) do
     Logger.info("Received bandwidth estimation of #{estimation / 1024} kbps")
+    estimation_increasing? = estimation >= state.available_bandwidth
 
     state =
       state
       |> Map.put(:available_bandwidth, estimation)
+      |> update_deficiency_statuses(estimation_increasing?)
       |> update_allocations()
+      |> update_probing_statuses()
       |> stop_probing_timer()
       |> start_probing_timer()
```

Comment:
Piping this looks pretty comoplicated. I assume that the order of these functions is important here, in other case we could just call `update_prober_status` but it is hard to get why it is so. These functions can also result in restarting timer 3 times in a row :thinking:

Reply:
Actually, we could use `update_prober_status` with a small modification to incorporate the logic of disallowing overuse, but we would end up with a *really* silly looking piece of code, roughly something like that:
```elixir
|> update_prober_status()
|> update_allocations()
|> update_prober_status()
...
```

Reply:
> These functions can also result in restarting timer 3 times in a row thinking

In theory, yes. However, I'm not sure this is such a big problem to warrant complicating the code even further with handling this case. Especially since this could happen in some really obscure corner cases and we're doing a full refactor anyway.